### PR TITLE
Handle Out-of-Footprint cases for PS1

### PR DESCRIPTION
**Pull request by profjsb**
This PR catches the case when the source/obj is not in the PS1 footprint, both for postage stamps and for the finding charts. It also sets a timeout (10 seconds) so that the stamp/image retrieval will not sit in limbo (possibly causing DB idle-in-transactions).

### CODE REVIEW

File: skyportal/utils/offset.py [hunk 1]
```diff
@@ -1026,6 +1026,9 @@ def fits_image(
             ra=center_ra, dec=center_dec, imsize=imsize
         )
 
+    if url in [None, ""]:
```
Comment (stefanv) [inline skyportal/utils/offset.py:1029]: Should this error hard, or just add an entry in the log?

**Review by stefanv — APPROVED** [review]

File: skyportal/models/obj.py [hunk 2]
```diff
@@ -481,15 +483,37 @@ def panstarrs_url(self):
         The cutout service doesn't allow directly querying for an image; the
         best we can do is request a page that contains a link to the image we
         want (in this case a combination of the g/r/i filters).
+
+        If this page does not return without PS1_CUTOUT_TIMEOUT seconds then
```
Comment (guynir42) [inline skyportal/models/obj.py:487]:
```suggestion
        If this page does not return within PS1_CUTOUT_TIMEOUT seconds then
```

File: skyportal/utils/offset.py [hunk 3]
```diff
@@ -42,6 +42,8 @@
 
 _, cfg = load_env()
 
+PS1_CUTOUT_TIMEOUT = 10
```
Comment (guynir42) [inline skyportal/utils/offset.py:45]:
```suggestion
from ..models.obj import PS1_CUTOUT_TIMEOUT
```

**Review by guynir42 — APPROVED** [review]
Minor suggestions below.

File: skyportal/utils/offset.py (hunk 1 above)
Comment (profjsb) [reply skyportal/utils/offset.py:1029]: it's ok to error here,  believe.

**Comment by profjsb** [comment]
ok @guynir42. If tests pass, should be good to merge.

**Pull request merged**